### DISCUSSION TITLE
How to properly extend telescope settings/mappings?

### DISCUSSION THREAD

**Discussion by AlgusDark** [LazyVim/LazyVim#940] [Q&A] [answered]
I'm trying:
```lua
{
  "nvim-telescope/telescope.nvim",
    opts = function(_, opts)
      opts = vim.tbl_extend("force", opts, {
        pickers = {
          live_grep = {
            path_display = { "shorten" },
            mappings = {
              i = {
                ["<C-a>"] = require("utils").live_grep_in_folder,
              },
            },
          },
        },
      })
      print(vim.inspect(opts)) -- It has the correct opts
    end,
}
```

The problem is that neither my mappings nor the `path_display` is being used by telescope.

If I do `opts = { pickers }` the mappings and settings are correct, but of course it will be lacking of the parent `opts` and it will break many things.

**Comment by AlgusDark** [comment]
Actually, doing the `opts = {pickers}` do the job. I had errors regarding other things.

**Comment by folke** [comment, marked as answer]
You never return the opts in that function, so they would never be used. Add a return opts

**Reply by AlgusDark** [reply]
Ah, silly of me. Thanks for the correct answer 🙏
